dictionarySearch.py

```diff
@@ -0,0 +1,27 @@
+"""Given a dictionary and a value, give list of all keys which have that value. In case of nesting, the keys should be separated by a dot.
+   {'a' : 'apple',
+     'b' : 'bobb',
+     'c' : {
+         'd' : 'dog'
+       },
+      'e' : 'dog'
+     }
+  I/P -> 'dog'
+  O/P -> ['e', 'c.d'] """
+
+d = {'a' : 'apple','b' : 'bobb','c' : {'d' : {'f':'dog'}},'e' : {'g':'dog'}}
+
+
+def value_search(value: int, dictionary: dict, matches=[], temp=""):
+    for key in dictionary:
```

Comment:
you end up doing `dictionary[key]:` throughout this code, just unpack it here

`for key, stored_value in dictionary.items():`